book/src/gobject_values.md

```diff
@@ -5,25 +5,24 @@ Enums in C are not as powerful as the one in Rust, which is why [`glib::Value`](
 Conceptually though, there are similar to a Rust enum defined like this:
 
 ```rust
-enum Value <T> {
-    bool(bool),
-    i8(i8),
-    i32(i32),
-    u32(u32),
-    i64(i64),
-    u64(u64),
-    f32(f32),
-    f64(f64),
-    // boxed types
-    String(Option<String>),
-    Object(Option<T: impl IsA<Object>>),
+enum Value {
+    bool,
+    i8,
+    i32,
+    u32,
+    i64,
+    u64,
+    f32,
+    f64,
+    String,
+    Object,
```

Comment:
The old thing was correct, with this you can't really hold any value

Reply:
Yeah, I know.
I wanted to remove it in order to not introduce additional complexity.
Probably it is more confusing now, will change it back.

Reply:
I want to get rid of the Option though, any objections?
@sdroege @bilelmoussaoui

Reply:
Why? You can still store an `Option<String>` and that's what it is internally, but you can get a `String` from it that simply returns an `Err` if it happens to be `None`.

Reply:
I still feel this is detail because of the implementation in C.
You will not want to store Option<String> more often than Option<i32>.
But you only can store Option<String>, so this is definitely surprising.

Just mentioning String and Object seems to cover most use cases and the rest can be handled by the docs in my opinion.

Reply:
Maybe just skip `String` and `Object` then to not confuse :)

Reply:
I will need them later though :)
I pushed another commit which re-adds the original setting and explains it a bit more.